dnf/comps.py

```diff
@@ -465,8 +465,18 @@ def _environment_install(self, env_id, pkg_types, exclude, strict=True):
         env = self.comps._environment_by_id(env_id)
         p_env = self.persistor.environment(env_id)
         if p_env.installed:
-            raise CompsError(_("Environment '%s' is already installed.") %
-                             env.ui_name)
+            logger.warning(_("Environment '%s' is already installed.") %
+                           env.ui_name)
+            if pkg_types in [MANDATORY, MANDATORY + CONDITIONAL,
+                             MANDATORY + OPTIONAL,
+                             MANDATORY + DEFAULT + OPTIONAL,
+                             MANDATORY + DEFAULT + CONDITIONAL + OPTIONAL]:
+                pass
+            elif pkg_types in [MANDATORY + DEFAULT,
+                               MANDATORY + DEFAULT + CONDITIONAL]:
+                pkg_types = pkg_types - DEFAULT
+            else:
+                raise ValueError(_('Unsupported group types combination'))
```

Comment:
@mluscon I did some minor changes but I think I didn't touch your comment. Somehow I didn't understand it.